src/bootstrap/builder.rs

```diff
@@ -1169,6 +1157,8 @@ impl<'a> Builder<'a> {
 
         self.ci_env.force_coloring_in_ci(&mut cargo);
 
+        cargo.env("RUSTFLAGS", &rustflags.0);
```

Comment:
Hm, so the problem I ran into with the previous attempt (moving the lints around) was that some of our CI scripts use `CARGO_TARGET_{}_RUSTFLAGS` if I got the order right and currently cargo will bail out completely if it finds `RUSTFLAGS` in the environment and just use that. Maybe we can use the target rustflags here? Not sure.

Reply:
Ah yeah I was attempting to handle that with [this block of code](https://github.com/rust-lang/rust/blob/72a79a099b2cd224a2a5f54f59bf4403a8ef3b6b/src/bootstrap/builder.rs#L1346-L1353) if it doesn't work out though I can try to work around it, haven't actually run the fuchsia builder yet.